### PR TITLE
Write post about 7.1.0.beta1 release

### PR DESCRIPTION
**Pull request by rafaelfranca**

### CODE REVIEW

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+user = User.new(name: "rafael", password: "railsrocks", password_confirmation: "railsrocks")
+user.save                                                                      # => true
+user.update(password: "pwn3d", password_challenge: "")                         # => false, challenge doesn't authenticate
+user.update(password: "railsGetsEvenBetter", password_challenge: "railsrocks") # => true
+```
+
+## A more robust framework
+
+Rails promises to be a framework that can get you from Hello World to IPO, and on each release we are improving
```
Comment (rafaelfranca) [inline _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:87]: @dhh I tried to see if we wrote a blog post about the "Hello World to IPO" but I can only find your tweets. Do you have any post in your blog about this that we can link here?

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+date: 2023-09-11 21:00:00 -04:00
+---
+
+RailsWorld is going to be soon, and to prepare for the celebration of the 20th anniversary of Rails, we are releasing the first beta of Rails 7.1.0!
+
+There are [over five thousand commits](https://github.com/rails/rails/compare/v7.0.8...main) since Rails 7.0, so this release is packed with new features and improvements.
+
+Please help us test all this new stuff so we can ensure a solid final release of Rails 7.1.
+
+## Generate Dockerfile for new applications
```
Comment (siaw23) [inline _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:16]:
```suggestion
## Generate Dockerfiles for new applications
```

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown [hunk 3]
```diff
@@ -0,0 +1,143 @@
+---
+layout: post
+title: 'Rails 7.1.0 Beta 1: Dockerfiles, Attribute Normalization, Single Use Tokens and More'
+categories: releases
+author: rafaelfranca
+published: true
+date: 2023-09-11 21:00:00 -04:00
+---
+
+RailsWorld is going to be soon, and to prepare for the celebration of the 20th anniversary of Rails, we are releasing the first beta of Rails 7.1.0!
```
Comment (siaw23) [inline _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:10]:
```suggestion
RailsWorld is approaching., and to prepare for the celebration of the 20th anniversary of Rails, we are releasing the first beta of Rails 7.1.0!
```

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+First, [`normalizes`](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Normalization/ClassMethods.html#method-i-normalizes)
+declares an attribute normalization. This is useful to normalize attributes before saving them to the database.
+
+```ruby
+class User < ApplicationRecord
+  normalizes :email, with: -> email { email.strip.downcase }
+end
+
+user = User.create(email: " CRUISE-CONTROL@EXAMPLE.COM\n")
```
Comment (siaw23) [inline _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:36]:
```suggestion
user = User.create(email: "CRUISE-CONTROL@EXAMPLE.COM\n")
```

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+There are [over five thousand commits](https://github.com/rails/rails/compare/v7.0.8...main) since Rails 7.0, so this release is packed with new features and improvements.
+
+Please help us test all this new stuff so we can ensure a solid final release of Rails 7.1.
+
+## Generate Dockerfile for new applications
+
+Rails will now generate a Docker-related files when you run `rails new`.
+
+These files serve as a foundational setup for deploying your Rails application in a
```
Comment (siaw23) [inline _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:20]:
```suggestion
These files serve as foundational setup for deploying your Rails application in a
```

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+The [`perform_all_later` method in Active Job](https://edgeapi.rubyonrails.org/classes/ActiveJob.html#method-c-perform_all_later),
+was designed to streamline the process of enqueuing multiple jobs simultaneously. This powerful
+addition allows you to efficiently enqueue jobs without triggering callbacks. This is
+particularly useful when you need to enqueue a batch of jobs at once, reducing the overhead
+of multiple round-trips to the queue datastore.
+
+Support to Composite Primary keys was added to Active Record, [allowing applications to improve the performance of some
+queries](https://shopify.engineering/how-to-introduce-composite-primary-keys-in-rails), specially those applications
+that use multi-tenant architecture, by using the tenant id as part of the primary key.
```
Comment (siaw23) [inline _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:101]:
```suggestion
that use multi-tenant architecture, by using the tenant ID as part of the primary key.
```

or

```suggestion
that use multi-tenant architecture, by using the tenant `id` as part of the primary key.
```

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+A new database adapter was added, [Trilogy](https://github.blog/2022-08-25-introducing-trilogy-a-new-database-adapter-for-ruby-on-rails/),
+a MySQL-compatible database client. This adapter is mostly a drop-in replacement for the MySQL2 adapter, but streamlines
+the installation process by not requiring the mysql client library to
+be present in the operation system, and improve the performance of some queries.
```
Comment (siaw23) [inline _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:106]:
```suggestion
be present in the operating system, and improve the performance of some queries.
```

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown (hunk 4 above)
Comment (rafaelfranca) [reply _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:36]: Space if by design. We are showing the strip call

Comment (siaw23) [reply _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:36]: Aaaaaa I missed that! 👍🏾

File: _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown (hunk 6 above)
Comment (nvasilevski) [reply _posts/2023-09-11-Rails-7-1-0-beta-1-has-been-released.markdown:101]: Agree with this suggestion! I'd say the second option could be `tenant_id` if we want to imply a particular column name or "tenant ID" if we want to imply "tenant identifier"

**Pull request merged**